osu.Game/Beatmaps/Drawables/DifficultyIcon.cs

```diff
@@ -66,4 +77,53 @@ private void load()
             };
         }
     }
+
+    public class DifficultyIconWithCounter : Container
+    {
+        private readonly List<BeatmapInfo> beatmaps;
+        private readonly OsuSpriteText text;
+        private readonly DifficultyIcon icon;
+
+        protected List<BeatmapInfo> Beatmaps
+        {
+            set
+            {
```

Comment:
why is this implementation in a protected setter that is seemingly only called from one place?

Reply:
`FilterableDifficultyIconWithCounter` inherits this class and it uses this property for filtering beatmaps
https://github.com/ppy/osu/blob/1bc87990968f8e71d1d33fe781d6e1cbbd022303/osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs#L226-L252